src/Core/NotificationCenter/Commands/CreateNotificationCommand.cs

```diff
@@ -31,6 +35,10 @@ public async Task<Notification> CreateAsync(Notification notification)
         await _authorizationService.AuthorizeOrThrowAsync(_currentContext.HttpContext.User, notification,
             NotificationOperations.Create);
 
-        return await _notificationRepository.CreateAsync(notification);
+        var newNotification = await _notificationRepository.CreateAsync(notification);
```

Comment:
@withinfocus For context, here is the current implementation of the "push notification" that has a 1-1 relation with `Notification` entity being created and a "push notification" being sent immediately afterwards.

Since the task system will now be creating many `Notification` entities for a user at once per our discussion, should we have a separate `BulkCreateNotificationCommand` that can send a new type of "push payload" that informs the user they have multiple new notifications? Or did you have some other type of interface / command structure in mind?

Reply:
Line 40 below sends a push for each notification, yes. You could make that optional via a parameter somehow here and reuse. I'd have to take a look at the bulk task creation logic but will that be using CQRS? I'd think you'd want to do a bulk insert of notifications too as you're alluding to, and then have one `PushSyncNotificationAsync` with some kind of "many notifications" payload.

Reply:
Yep, there is a bulk task creation CQRS command [under review here](https://github.com/bitwarden/server/pull/5188/files#diff-b90aca4e88c21fa0ff26d53076a322a0e323ce035f83767e56d2e47e523caf42). 

We're hooking into this `CreateNotificationCommand` [here](https://github.com/bitwarden/server/pull/5344/files/b43b13107b53ab275338e55b84ca88e461b9f46c). We could swap over to a new `BullkCreateNotificationCommand` there with the behavior we've described -- Bulk insert notifications and one `PushSyncNotificationAsync` with a "many notifications" payload.